### PR TITLE
Add demo seed task

### PR DESCRIPTION
**Pull request by chunky56**
To get the demo seed data to work for a build, run the two commands:
gradle clean setupdb seed build
gradle setupdb baseseed demoseed run

### CODE REVIEW

File: modules/db/build.gradle [hunk 1]
```diff
@@ -119,13 +124,23 @@ task baseSeed(type: Exec) {
     commandLine 'psql', '-U', dbUser, '--file', 'src/main/resources/seed/seed.sql', '-h', databaseHostName, '-w', dbName
 }
 
-task enableCustom(type: Exec, dependsOn: baseSeed) {
+task extraSeed(type: Exec, dependsOn: baseSeed) {
```
Comment (richmag) [inline modules/db/build.gradle]: Do we have any doc on the intent of the various seed files and tasks?

**Comment by richmag** [comment]
Looks good (though I did not test loading it yet).  I think we'd need to update the readme to mention the critical seed tasks.  Somewhere we'd need to doc the users added, their passwords, etc.  Not sure if a file or a wiki page is best for that.

**Comment by benleibert** [comment]
I'm also looking through this, and will have a couple of questions to ask before merging.

File: modules/db/build.gradle (hunk 1 above)
Comment (chunky56) [inline modules/db/build.gradle]: We could put descriptions in each task to describe what they do.

**Comment by benleibert** [comment]
Hi Chongsun. Like Rich, I was going to ask about adding comments... which you’ve largely done in the most recent commit. It would still be nice, though, to tweak/add comments to some of the SQL files. For instance, the test-data/sql/demo_seed.sql file starts with a lengthy comment. It would be really helpful, though, if it described what open_lmis_baseseed and open_lmis_demoseed are . (They're obviously databases, but what's in them, and how do they differ?) Some of the other SQL files might similarly benefit. (eg: a comment at the top of true_seed.sql, describing what it gives you, and reminders at the top of the enable_custom.sql/disable_custome.sql files describing their purpose.)  

I also wonder about the use of dependsOn in the gradle file. Specifically, it looks as though the order in which its dependencies are run in significant for the “seed” task (and maybe others as well). In older versions of gradle, and I think perhaps the current one as well, there’s no guarantee that dependencies are run in the order specified within the dependsOn property. I haven’t fully researched this within the context of gradle v2, though did come across this: http://trickyandroid.com/gradle-tip-3-tasks-ordering/. If it isn’t an issue you’ve already considered, I suggest giving it a little thought in order to potentially avoid what could be a subtle and sporadic issue.

Finally, I wonder about the two commands you mentioned. Because setupDB drops the database, wouldn’t the setupDB task run as part of the second of the two commands delete the work done within the first of the two commands? Would the following (which doesn’t include baseseed) work, instead, to run all of the tasks at once, in order to build a database intended for use on our new test server:       gradle clean setupdb seed demoseed build run

**Comment by chunky56** [comment]
I would actually prefer to limit what is entered in the demo_seed.sql file, as it is generated and possible further iterations of it will probably be generated as well. If the (generated) comments are confusing (like what is open_lmis_baseseed and open_lmis_demoseed), I would probably just remove those comments altogether and just document it somewhere else (like in the gradle task).

**Comment by chunky56** [comment]
For the dependsOn, I ensure order by making the latter subtask dependent on the former. This is probably sufficient, and I probably only need to make the metatask dependent on the latter task, but I make the metatask dependent on both to explicitly show dependence, rather than enforcing order. If you have a better way to do this, I'm open to suggestions, as I'm not terribly familiar with the ins and outs of gradle.

**Comment by chunky56** [comment]
Yes, you need to run the gradle commands separately in order for it to work. The second setupdb does blow out the first setupdb and this is intentional. The first command is to build the war code. The second command is to get the database in the right state. The demoseed task depends on the database being in the state after baseseed has been applied, not after seed has been applied. This is because the demo_seed script is the diff between the db after the baseseed task, and the db after the baseseed and demoseed tasks have been applied. The reason why I do not build the demo data off of seed, is because there is extra data that would need to be deleted (samples facilities, geo zones, etc..).

I tried to find a way to get gradle clean setupdb seed build demoseed run, to work, but it does not work because of the nature of seed not being a true seed, but making it a true seed would require redoing all of the core integration tests.

**Comment by benleibert** [comment]
Thank you very much for explaining the need for the two commands. It would have taken a while to figure that out, and I wonder if your explanation should perhaps be put on the project’s main README.md file. Whenever you think the documentation is ready, I think the rest of the pull-request is ready as well.

**Comment by richmag** [comment]
Fine by me, too.

**Pull request merged**